docs/src/Modal.doc.js

```diff
@@ -23,85 +32,497 @@ card(
         type: 'string',
         required: true,
         description:
-          'String that clients such as VoiceOver will read to describe the modal. Always localize the label.',
-        href: 'accessibility',
+          'String that clients such as VoiceOver will read to describe the modal. Always localize the label. See the [Accessibility section](#Accessibility) for more info.',
       },
       {
         name: 'align',
         type: `"center" | "left"`,
         defaultValue: 'center',
-        description: 'Use to specify the alignment of `heading` & `subHeading` strings',
-        href: 'headingOptions',
+        description:
+          'Use to specify the alignment of `heading` & `subHeading` strings. See the [Heading variant](#Heading) for more info.',
       },
       {
         name: 'children',
         type: 'React.Node',
+        description:
+          "Supply the element(s) that will be used as Modal's main content. See the [Best Practices](#Best-practices) for more info.",
       },
       {
         name: 'closeOnOutsideClick',
         type: 'boolean',
-        description: 'Close the modal when you click outside of it',
+        description:
+          'Close the modal when you click outside of it. See the [outside click variant](#Preventing-close-on-outside-click) for more info.',
         defaultValue: true,
-        href: 'closeOnOutsideClickExample',
       },
       {
         name: 'footer',
         type: 'React.Node',
-        href: 'sizesExample',
+        description:
+          "Supply the container element(s) or render prop that will be used as Modal's custom footer. See the [Best Practices](#Best-practices) for more info.",
       },
       {
         name: 'heading',
         type: `string | React.Node`,
-        required: false,
-        href: 'heading',
+        description:
+          "The text used for Modal's heading. Be sure to localize this text. See the [Heading variant](#Heading) for more info.",
       },
       {
         name: 'onDismiss',
         type: '() => void',
         required: true,
-        href: 'sizesExample',
+        description:
+          'Callback fired when Modal is dismissed by clicking on the a dismiss-type Footer button or clicking on the backdrop outside of the Modal (if `closeOnOutsideClick` is true).',
       },
       {
         name: 'ref',
         type: "React.Ref<'div'>",
         description: 'Forward the ref to the underlying div element',
```

Comment:
This description makes it sound like `ref` is a boolean indicating if the user wants a ref forwarded. Consider something like "Ref that will set on the underlying `<div>` element."